### ISSUE TITLE
Readme instructions don't appear correct

### ISSUE THREAD

**Issue by ChrisMcKee** [algolia/algoliasearch-client-go#543]
Following the instructions in the readme verbaitm fails. 

Go 1.13 / ubuntu + windows.

```
$ go mod init playground
go: creating new go.mod: module playground

$ go get github.com/algolia/algoliasearch-client-go@v3.3.0

go: finding github.com v3.3.0
go: finding github.com/algolia/algoliasearch-client-go v3.3.0
go: finding github.com/algolia v3.3.0
go: finding github.com/algolia/algoliasearch-client-go v3.3.0
go get github.com/algolia/algoliasearch-client-go@v3.3.0: github.com/algolia/algoliasearch-client-go@v3.3.0: invalid version: module contains a go.mod file, so major version must be compatible: should be v0 or v1, not v3
```

**Comment by aseure** [comment]
Hello @ChrisMcKee, thank you very much for taking the time to report this issue.

The issue was already there before but cannot be mitigated anymore since Go 1.13 (I've been able to reproduce today with Go 1.13).

The origin of this issue is that when we introduced the use of Go modules in `v3.0.0`, we did not use the `v3/` prefix in module declarations. However, this seems mandatory for any published module with at least two major versions released [as documented in the golang wiki](https://github.com/golang/go/wiki/Modules#releasing-modules-v2-or-higher).

As for a solution, as of today, I cannot see any workaround that would not imply a breaking-change on the user-end. I've forked the repository to try to make a clean v3 release that would work with the `go get`/`go build` toolchain. On the user side, it means changing all import path from source files and the require statement of the `go.mod` file to append the needed `v3/` suffix. This can be done with a shell one-liner. We've been discussing it with the team internally and we are planning on introducing this breaking-change in a new minor version, along with the shell one-liner. I'm going to prepare a PR for that will wait for your feedback @ChrisMcKee.

cc @Ant-hem

**Comment by ChrisMcKee** [comment]
Sounds fine; I'd read the issue in a few other places and happily I've been around long enough to know how to get around it locally using normal version referencing. Just a good one to get right in the first thing people come across (readme being reflected in the docs as well) 

I'll happily give it a test when its ready; as I say, not a show-stopper but incorrect / broken as far as instructions go 👍

**Comment by aseure** [comment]
Closed by #544 and released as of [`v3.4.0`](https://github.com/algolia/algoliasearch-client-go/blob/master/ChangeLog.md#340-2019-09-25)
